### PR TITLE
Fixes for enkf archival bugs in ush/hpssarch_gen.sh

### LINKED ISSUE #1622: earc jobs not archiving expected files
**Expected behavior**
enkfgdasearc jobs should archive more than just enkfgdas log files and recenered analysis increment files.
 
**Current behavior**
A Hera parallel using current head of develop is only archiving enkfgdas log files in `enkfgdas.tar` and recentered analysis increments in `enkfgdas_grp*tar`.

**Machines affected**
Parallel run on Hera

**To Reproduce**
Install g-w and run a parallel with enkfgdas active.  Check output from earc jobs.

**Detailed Description**
`exgdas_enkf_earch.sh` executes `hpssarch_gen.sh` to create a list of paths/files to archive.   A quick check of `hpssarch_gen.sh` suggests that the script not fully updated to the new GFS COM directory structure.   

For example, the `enkfgdas` section of `hpssarch_gen.sh` has the following for member atmf and sfcf files
```
          echo "${COM_ATMOS_ANALYSIS_MEM/${ROTDIR}\//}/${head}atmf00${FHR}.nc"
          if (( FHR == 6 )); then
            echo "${COM_ATMOS_ANALYSIS_MEM/${ROTDIR}\//}/${head}sfcf00${FHR}.nc"
          fi
```
Member forecasts are in `COM_ATMOS_HISTORY`, not `COM_ATMOS_ANALYSIS_MEM`

### LINKED ISSUE #1623: Error in creation of enkf archive file list with DOIAU=YES
**Current behavior**

Generated list of enkf files to archive is wrong. This `enkfgdas.t00z.atmf0036,9.nc` is wrong.

**Machines affected**

All (any that archive to HPSS).

**To Reproduce**

Build and run global-workflow develop with `DOIAU=YES` and enkf archival turned on.

**Additional Information**

Log on Hera: /scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/verfiles/enkfgdasearc01.log.0

Error at bottom of log:
```
++ exgdas_enkf_earc.sh[69]: cat /scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/verfiles/enkfgdas.20220102/00/earc01/enkfgdas_grp1.txt
+ exgdas_enkf_earc.sh[69]: htar -P -cvf /NCEPDEV/emc-global/1year/Kate.Friedman/HERA/scratch/verfiles/2022010200/enkfgdas_grp01.tar enkfgdas.20220102/00/mem001/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem002/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem003/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem004/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem005/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem006/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem007/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem008/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem009/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem010/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc
[connecting to hpsscore1.fairmont.rdhpcs.noaa.gov/1217]
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem001/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem001/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem002/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem002/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem003/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem003/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem004/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem004/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem005/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem005/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem006/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem006/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem007/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem007/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem008/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem008/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem009/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem009/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
ERROR: /apps/hpss/bin/htar: error -1 stat-ing "enkfgdas.20220102/00/mem010/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc"
*** Warning: No matching names located for 'enkfgdas.20220102/00/mem010/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc'
INFO: refusing to create empty archive
HTAR: HTAR SUCCESSFUL
###WARNING  htar returned non-zero exit status.
            72 = /apps/hpss/bin/htar -P -cvf /NCEPDEV/emc-global/1year/Kate.Friedman/HERA/scratch/verfiles/2022010200/enkfgdas_grp01.tar enkfgdas.20220102/00/mem001/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem002/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem003/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem004/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem005/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem006/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem007/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem008/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem009/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc enkfgdas.20220102/00/mem010/analysis/atmos/enkfgdas.t00z.atmf0036,9.nc
+ exgdas_enkf_earc.sh[1]: postamble exgdas_enkf_earc.sh 1684276225 72
```

Traced list generation back to a list of output files designated by the contents of `IAUFHRS_ENKF`. When `DOIAU=YES` and `DOIAU_ENKF=YES` the value of `IAUFHRS_ENKF` is `3,6,9`.

```
+++ config.base[269]: export DOIAU=YES
+++ config.base[269]: DOIAU=YES
+++ config.base[270]: export IAUFHRS=3,6,9
+++ config.base[270]: IAUFHRS=3,6,9
++++ config.base[271]: echo 3,6,9
++++ config.base[271]: cut -c1
+++ config.base[271]: export IAU_FHROT=3
+++ config.base[271]: IAU_FHROT=3
+++ config.base[272]: export IAU_DELTHRS=6
+++ config.base[272]: IAU_DELTHRS=6
+++ config.base[273]: export IAU_OFFSET=6
+++ config.base[273]: IAU_OFFSET=6
+++ config.base[274]: export DOIAU_ENKF=YES
+++ config.base[274]: DOIAU_ENKF=YES
+++ config.base[275]: export IAUFHRS_ENKF=3,6,9
+++ config.base[275]: IAUFHRS_ENKF=3,6,9
```

Later, while in `hpssarch_gen.sh`...

```
+ hpssarch_gen.sh[472](enkfgdas): IAUFHRS_ENKF=3,6,9
```
...the value of `nfhrs` becomes:
```
+ hpssarch_gen.sh[474](enkfgdas): nfhrs=36,9
```
This should likely be `369` instead of `36,9` or a list with spaces (?) `3 6 9` for a for-loop later.

In `hpssarch_gen.sh`, line 474:
```
468 #-----------------------------------------------------
469 if [[ ${type} == "enkfgdas" || ${type} == "enkfgfs" ]]; then
470 #-----------------------------------------------------
471 
472   IAUFHRS_ENKF=${IAUFHRS_ENKF:-6}
473   lobsdiag_forenkf=${lobsdiag_forenkf:-".false."}
474   nfhrs="${IAUFHRS_ENKF/,/}"
...
501     for FHR in ${nfhrs}; do  # loop over analysis times in window
```
...should probably be:
```
474   nfhrs="${IAUFHRS_ENKF//,}"
```
Change the order of the 2nd forward slash and comma.

Tested this on the command line:
```
-bash-4.2$ IAUFHRS_ENKF="3,6,9"
-bash-4.2$ echo $IAUFHRS_ENKF
3,6,9
-bash-4.2$ echo "${IAUFHRS_ENKF/,/}"
36,9
-bash-4.2$ echo "${IAUFHRS_ENKF//,}"
369
```

**Possible Implementation**

Looking into this further for possible resolution.

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
**Description**

This PR brings in fixes to `ush/hpssarch_gen.sh` for enkf archival bugs described in issues #1622 and #1623 and beyond:

- Use the correct path for ensemble atm and sfc files (these are located in atm history, not atm analysis
- Fix a syntax error to properly loop over IAU hours
- Update the name of the 'log' files

Fixes #1622
Fixes #1623

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**

- Reran failing earc jobs in test on Hera (DOIAU=YES and DOIAU=NO) (@KateFriedman-NOAA & @RussTreadon-NOAA)
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: ush/hpssarch_gen.sh [hunk 1]
```diff
@@ -560,11 +560,12 @@ if [[ ${type} == "enkfgdas" || ${type} == "enkfgfs" ]]; then
 
       MEMDIR="mem${mem}" YMD=${PDY} HH=${cyc} generate_com \
         COM_ATMOS_ANALYSIS_MEM:COM_ATMOS_ANALYSIS_TMPL \
-        COM_ATMOS_RESTART_MEM:COM_ATMOS_RESTART_TMPL
+        COM_ATMOS_RESTART_MEM:COM_ATMOS_RESTART_TMPL \
+        COM_ATMOS_HISTORY:COM_ATMOS_HISTORY_TMPL
```
Comment (WalterKolczynski-NOAA) [inline ush/hpssarch_gen.sh]:
```suggestion
        COM_ATMOS_HISTORY_MEM:COM_ATMOS_HISTORY_TMPL
```

File: ush/hpssarch_gen.sh [hunk 2]
```diff
@@ -598,9 +599,9 @@ if [[ ${type} == "enkfgdas" || ${type} == "enkfgfs" ]]; then
           fi
         fi
         {
-          echo "${COM_ATMOS_ANALYSIS_MEM/${ROTDIR}\//}/${head}atmf00${FHR}.nc"
+          echo "${COM_ATMOS_HISTORY/${ROTDIR}\//}/${head}atmf00${FHR}.nc"
```
Comment (WalterKolczynski-NOAA) [inline ush/hpssarch_gen.sh]:
```suggestion
          echo "${COM_ATMOS_HISTORY_MEM/${ROTDIR}\//}/${head}atmf00${FHR}.nc"
```

File: ush/hpssarch_gen.sh [hunk 3]
```diff
@@ -598,9 +599,9 @@ if [[ ${type} == "enkfgdas" || ${type} == "enkfgfs" ]]; then
...
           if (( FHR == 6 )); then
-            echo "${COM_ATMOS_ANALYSIS_MEM/${ROTDIR}\//}/${head}sfcf00${FHR}.nc"
+            echo "${COM_ATMOS_HISTORY/${ROTDIR}\//}/${head}sfcf00${FHR}.nc"
```
Comment (WalterKolczynski-NOAA) [inline ush/hpssarch_gen.sh]:
```suggestion
            echo "${COM_ATMOS_HISTORY_MEM/${ROTDIR}\//}/${head}sfcf00${FHR}.nc"
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: jobs/JGDAS_ENKF_ARCHIVE [hunk 4]
```diff
@@ -12,6 +12,7 @@ export CDUMP=${RUN/enkf}
 YMD=${PDY} HH=${cyc} generate_com -rx COM_TOP
 MEMDIR="ensstat" YMD=${PDY} HH=${cyc} generate_com -rx \
   COM_ATMOS_ANALYSIS_ENSSTAT:COM_ATMOS_ANALYSIS_TMPL
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGDAS_ENKF_ARCHIVE]:
```suggestion
  COM_ATMOS_ANALYSIS_ENSSTAT:COM_ATMOS_ANALYSIS_TMPL \
```

Comment (KateFriedman-NOAA) [reply jobs/JGDAS_ENKF_ARCHIVE]: Works in testing. Accepting commit suggestion.

File: ush/hpssarch_gen.sh (hunk 1 above)
Comment (KateFriedman-NOAA) [reply ush/hpssarch_gen.sh]: Works in testing. Accepting commit suggestion.

File: ush/hpssarch_gen.sh (hunk 2 above)
Comment (KateFriedman-NOAA) [reply ush/hpssarch_gen.sh]: Works in testing. Accepting commit suggestion.

File: ush/hpssarch_gen.sh (hunk 3 above)
Comment (KateFriedman-NOAA) [reply ush/hpssarch_gen.sh]: Works in testing. Accepting commit suggestion.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
@RussTreadon-NOAA Other than choking on the files UFSDA doesn't produce (which we'll tackle later), is everything working now with these changes?

**Comment by RussTreadon-NOAA** [comment]
@WalterKolczynski-NOAA, I have only tested using a UFSDA ATM var & ens parallel.  I don't have an active GSI-EnKF parallel to test with.

**Pull request merged**